src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -1463,17 +1432,19 @@ public boolean hasCitizenCompletedRequests(@NotNull final CitizenData data)
         return !getCompletedRequests(data).isEmpty();
     }
 
+    @SuppressWarnings(RAWTYPES)
     public ImmutableList<IRequest> getCompletedRequests(@NotNull final CitizenData data)
     {
-        if (!citizensByCompletedRequests.containsKey(data.getId()))
+        if (!getCompletedRequestsByCitizen().containsKey(data.getId()))
         {
             return ImmutableList.of();
         }
 
-        return ImmutableList.copyOf(citizensByCompletedRequests.get(data.getId()).stream()
+        return ImmutableList.copyOf(getCompletedRequestsByCitizen().get(data.getId()).stream()
                                       .map(getColony().getRequestManager()::getRequestForToken).filter(Objects::nonNull).iterator());
     }
 
+    @SuppressWarnings({GENERIC_WILDCARD, RAWTYPES, UNCHECKED})
     public <R> ImmutableList<IRequest<? extends R>> getCompletedRequestsOfType(@NotNull final CitizenData citizenData, final TypeToken<R> requestType)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)